### ISSUE TITLE
bug(resources): not working!

### ISSUE THREAD

**Issue by dylanvdmerwe** [driftyco/ionic-cli#1534]
#### Short description of the problem:
Running `ionic resources` no longer works!

Icons are not being generated on Mac. Splash screen images are not being generated on Mac. 

#### What behaviour are you expecting?

- [ ] Running the `ionic resources` command generates icons for iOS
- [ ] Running the `ionic resources` command generates icons for Android
- [ ] Running the `ionic resources` command generates splash screens for iOS
- [ ] Running the `ionic resources` command generates splash screens for Android
- [ ] Running the `ionic resources` command updates config.xml file with generated images (and icons) 
- [ ] Resources in the config file must work on Mac and Windows irrespective of the structure of the filepath. https://github.com/driftyco/ionic-cli/issues/1408

In my /resources folder I have the following:
-- [resources]
&nbsp;&nbsp;&nbsp;-- [android]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-- [icon]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-- [splash]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-- icon.png ** specific icon for android
&nbsp;&nbsp;&nbsp;-- [ios]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-- [icon]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-- [splash]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-- icon.png ** specific icon for ios
&nbsp;&nbsp;&nbsp;-- icon.png ** default icon image provided by ionic
&nbsp;&nbsp;&nbsp;-- splash.png ** customised splash screen for both ios and android

What has happened to this functionality?!

Output from when I run `ionic resources` when I have both ios and android platforms added:
```
$ ionic resources
Ionic icon and splash screen resources generator
 uploading android/icon.png...
 uploading ios/icon.png...
 uploading splash.png...
ios/icon.png (192x192) upload complete
android/icon.png (192x192) upload complete
splash.png (2732x2732) upload complete
 generating splash ios Default~iphone.png (320x480)...
 generating splash ios Default@2x~iphone.png (640x960)...
 generating splash ios Default-Portrait~ipad.png (768x1024)...
splash ios Default@2x~iphone.png (640x960) generated
 generating splash ios Default-Portrait@~ipadpro.png (2048x2732)...
splash ios Default-Portrait~ipad.png (768x1024) generated
 generating splash ios Default-Portrait@2x~ipad.png (1536x2048)...
splash ios Default~iphone.png (320x480) generated
 generating splash ios Default-Landscape~ipad.png (1024x768)...
splash ios Default-Landscape~ipad.png (1024x768) generated
 generating splash ios Default-Landscape@~ipadpro.png (2732x2048)...
splash ios Default-Portrait@2x~ipad.png (1536x2048) generated
 generating splash ios Default-Landscape@2x~ipad.png (2048x1536)...
splash ios Default-Landscape@2x~ipad.png (2048x1536) generated
 generating splash ios Default-Landscape-736h.png (2208x1242)...
splash ios Default-Landscape-736h.png (2208x1242) generated
 generating splash ios Default-736h.png (1242x2208)...
splash ios Default-736h.png (1242x2208) generated
 generating splash ios Default-667h.png (750x1334)...
splash ios Default-667h.png (750x1334) generated
 generating splash ios Default-568h@2x~iphone.png (640x1136)...
splash ios Default-568h@2x~iphone.png (640x1136) generated
 generating splash android drawable-port-xxxhdpi-screen.png (1280x1920)...
splash android drawable-port-xxxhdpi-screen.png (1280x1920) generated
 generating splash android drawable-port-xxhdpi-screen.png (960x1600)...
splash android drawable-port-xxhdpi-screen.png (960x1600) generated
 generating splash android drawable-port-xhdpi-screen.png (720x1280)...
splash android drawable-port-xhdpi-screen.png (720x1280) generated
 generating splash android drawable-port-hdpi-screen.png (480x800)...
splash android drawable-port-hdpi-screen.png (480x800) generated
splash android drawable-port-mdpi-screen.png (320x480) generated
 generating splash android drawable-port-ldpi-screen.png (240x320)...
splash android drawable-port-ldpi-screen.png (240x320) generated
 generating splash android drawable-land-xxxhdpi-screen.png (1920x1280)...
splash android drawable-land-xxxhdpi-screen.png (1920x1280) generated
 generating splash android drawable-land-xxhdpi-screen.png (1600x960)...
splash android drawable-land-xxhdpi-screen.png (1600x960) generated
 generating splash android drawable-land-xhdpi-screen.png (1280x720)...
splash android drawable-land-xhdpi-screen.png (1280x720) generated
 generating splash android drawable-land-hdpi-screen.png (800x480)...
splash android drawable-land-hdpi-screen.png (800x480) generated
 generating splash android drawable-land-mdpi-screen.png (480x320)...
splash android drawable-land-mdpi-screen.png (480x320) generated
 generating splash android drawable-land-ldpi-screen.png (320x240)...
splash android drawable-land-ldpi-screen.png (320x240) generated
 generating icon ios icon-small@3x.png (87x87)...
icon ios icon-small@3x.png (87x87) generated
 generating icon ios icon-small@2x.png (58x58)...
icon ios icon-small@2x.png (58x58) generated
 generating icon ios icon-small.png (29x29)...
icon ios icon-small.png (29x29) generated
 generating icon ios icon-83.5@2x.png (167x167)...
icon ios icon-83.5@2x.png (167x167) generated
 generating icon ios icon-76@2x.png (152x152)...
icon ios icon-76@2x.png (152x152) generated
 generating icon ios icon-76.png (76x76)...
icon ios icon-76.png (76x76) generated
 generating icon ios icon-72@2x.png (144x144)...
icon ios icon-72@2x.png (144x144) generated
 generating icon ios icon-72.png (72x72)...
icon ios icon-72.png (72x72) generated
 generating icon ios icon-60@3x.png (180x180)...
icon ios icon-60@3x.png (180x180) generated
 generating icon ios icon-60@2x.png (120x120)...
icon ios icon-60@2x.png (120x120) generated
 generating icon ios icon-60.png (60x60)...
icon ios icon-60.png (60x60) generated
 generating icon ios icon-50@2x.png (100x100)...
icon ios icon-50@2x.png (100x100) generated
 generating icon ios icon-50.png (50x50)...
icon ios icon-50.png (50x50) generated
icon ios icon-40@3x.png (120x120) generated
 generating icon ios icon-40@2x.png (80x80)...
icon ios icon-40@2x.png (80x80) generated
 generating icon ios icon-40.png (40x40)...
icon ios icon-40.png (40x40) generated
 generating icon ios icon@2x.png (114x114)...
icon ios icon@2x.png (114x114) generated
 generating icon ios icon.png (57x57)...
icon ios icon.png (57x57) generated
 generating icon android drawable-xxxhdpi-icon.png (192x192)...
icon android drawable-xxxhdpi-icon.png (192x192) generated
 generating icon android drawable-xxhdpi-icon.png (144x144)...
icon android drawable-xxhdpi-icon.png (144x144) generated
 generating icon android drawable-xhdpi-icon.png (96x96)...
icon android drawable-xhdpi-icon.png (96x96) generated
 generating icon android drawable-hdpi-icon.png (72x72)...
icon android drawable-hdpi-icon.png (72x72) generated
 generating icon android drawable-mdpi-icon.png (48x48)...
icon android drawable-mdpi-icon.png (48x48) generated
 generating icon android drawable-ldpi-icon.png (36x36)...
icon android drawable-ldpi-icon.png (36x36) generated
```

**Post the output of `ionic info` below please**
Cordova CLI: 6.3.1
Gulp version:  CLI version 3.9.1
Gulp local:
Ionic Framework Version: 2.0.0-rc.0
Ionic CLI Version: 2.1.0
Ionic App Lib Version: 2.1.0-beta.1
ios-deploy version: 1.8.6
ios-sim version: 5.0.8
OS: Mac OS X El Capitan
Node Version: v5.12.0
Xcode version: Xcode 7.3.1 Build version 7D1014

**Comment by dylanvdmerwe** [comment]
Not sure if this is working on Windows. My colleagues seem to think so.

**Comment by mlynch** [comment]
So, are you seeing any files in, say `resources/ios/icon`?

**Comment by dylanvdmerwe** [comment]
When I ran the command above, no. Nothing was generated even though it said it was. The original ionic starter images remained in their respective folders and not overridden.

**Comment by mlynch** [comment]
What I mean to say is, is the issue that `ionic resources` isn't generating updated versions of your icons, or is it not generating even the default ones? For example, running `ionic resources` on a fresh Ionic project gives me these icons:

<img width="1065" alt="screenshot 2016-10-20 16 03 46" src="https://cloud.githubusercontent.com/assets/11214/19577902/d02345b4-96de-11e6-9f19-1bb0f56e6174.png">

**Comment by dylanvdmerwe** [comment]
As a test, I have just deleted all images from android/icon, android/splash, ios/icon, ios/splash and run the `ionic resources` command again and no images were generated at all.

**Comment by mlynch** [comment]
Hmm. I'm testing it on my end and it's working fine. Can you email me the file you're using for a logo/splash? Maybe it's related to that? max@ionic.io

**Comment by dylanvdmerwe** [comment]
I'll drop you a mail shortly.

**Comment by mlynch** [comment]
Thanks for the info. I can reproduce after a few steps it seems to get into an inconsistent state. Looking into a fix right now.

**Comment by dylanvdmerwe** [comment]
Much appreciated Max! Would be a life saver. Note that [this](https://github.com/driftyco/ionic-cli/issues/1408) issue is also linked to this.

**Comment by mlynch** [comment]
Commit to fix https://github.com/driftyco/ionic-app-lib/commit/2cd46b5b6e7fb2c28aa7409b5d07074da1cc4f62

**Comment by mlynch** [comment]
Okay, so quick update. There was an issue on the server with the iPad pro sizes which I just pushed and reloaded the server, so that side is fixed.

On the CLI side, can you try running ionic resources again to see if it'll work for you now that the server is fixed?

If that doesn't work, can you try this as a fix until we get get a new release out?

```
cd /var/folders/hd
find ./ -name "*320x480*"
```
You should see a folder here that has your tmp files:
<img width="459" alt="screenshot 2016-10-20 17 32 21" src="https://cloud.githubusercontent.com/assets/11214/19580350/48b58968-96eb-11e6-8073-4b1a846c6903.png">

Then run
```
open ctkkc94j2wg9mkgb28w413980000gn/T
```

And delete all the images that match the ones you have in your splash/icon

Then try re-running and it Should Work.

**Comment by dylanvdmerwe** [comment]
Hi Max. It appears that the changes you made on the server sorted everything out. Things are now generating. 

Will this also fix the path issues between Windows and Mac for resources as described on this issue? https://github.com/driftyco/ionic-cli/issues/1408

**Comment by jthoms1** [comment]
@dylanvdmerwe Just released version 2.1.4 of the CLI.  Could you give it a try?

**Comment by jthoms1** [comment]
This should now be resolved with the latest CLI release.
